typescript/agentkit/src/wallet-providers/solanaKeypairWalletProvider.ts

```diff
@@ -57,10 +56,14 @@ export class SolanaKeypairWalletProvider extends SvmWalletProvider {
   }) {
     super();
 
-    this.#keypair =
-      typeof keypair === "string"
-        ? Keypair.fromSecretKey(bs58.decode(keypair))
-        : Keypair.fromSecretKey(keypair);
+    if (typeof keypair === "string") {
+      import("bs58").then(bs58 => {
```

Comment:
pnpm strictly respects package.json type fields and module/main exports, while npm was more lenient with ESM/CommonJS interop. Refactoring npm to pnpm had the side effect of pointing out that bs58 had such issue.

I'm not a fan of the fix though. We use this in the constructor, meaning we cannot await the dynamic import. This also means we cannot ensure #keypair is setup, and thus it needs to be nullable.

I am happy to discuss a cleaner fix, or to fast-follow with one

Reply:
This has been resolved and this bandaid fix reverted. Thank you @0xRAG